xscen/catalog.py

```diff
@@ -411,6 +411,14 @@ def create(
         return cls(str(meta_path))
 
     def __init__(self, df, *args, **kwargs):
+        if "create" in kwargs:
+            kwargs.pop("create")
+            if isinstance(df, (str, Path)) and (
+                not os.path.isfile(Path(df)) or kwargs.get("overwrite", None) is True
+            ):
+                self.create(df, **kwargs)
+        kwargs.pop("project", None)
+        kwargs.pop("overwrite", None)
```

Comment:
```suggestion
    def __init__(self, df, *args, create=False, overwrite=None, project=None, **kwargs):
        if create:
            if isinstance(df, (str, Path)) and (not os.path.isfile(Path(df)) or overwrite):
                self.create(df, project=project, overwrite=overwrite)
```
If kwargs are not passed directly to a parent, I find it makes more sense to have them explicitly in the signature.

But in fact I am wondering if your signatures make sense at all here. When I dig, I can't see what `*args` could be and the only meaningful `kwarg` are `read_csv_kwargs` and `registry`. Why not list them directly? I can push something if you agree.

Reply:
(outdated comment, but I think it addresses your last changes too)

Reply:
I'm honestly not sure either... I think we might just have used `*args, **kwargs` as a default way to support whatever changes could occur in `intake-esm`?

Reply:
Indeed, it's a good idea to be ready for future changes... In that case, then I think this is satisfactory!